clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -89,18 +89,26 @@ public void append(long offset, Record record) {
     /**
      * Append a new record and offset to the buffer
      */
-    public void append(long offset, long timestamp, byte[] key, byte[] value) {
+    public void append(long offset, long timestamp, long checksum, byte[] key, byte[] value) {
```

Comment:
Not a big deal, but the inconsistent usage of 'checksum' and 'crc' is a little annoying.